Classes/AccessControl/Request.php

```diff
@@ -129,8 +129,23 @@ public function __construct(array $environment) {
       $this->origin = new Uri($environment['HTTP_ORIGIN']);
 
       $this->isCrossOrigin = $this->origin->getScheme() != $this->destination->getScheme() ||
-        $this->origin->getHostname() != $this->destination->getHostname() ||
-        $this->origin->getPort() != $this->destination->getPort();
+        $this->origin->getHostname() != $this->destination->getHostname();
+
+      // compare ports.
+      if ($this->isCrossOrigin === FALSE && $this->origin->getPort() != $this->destination->getPort()) {
```

Comment:
Can you move this before the check above so that we have a normalized destination and origin port which could be used in the check?